geo: fix to_json representation of GeoJSON

otan opened the pull request:
Fixed the to_json() representation of geospatial objects to be the
same as the GeoJSON format.

Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/52684)
<!-- Reviewable:end -->

sumeerbhola reviewed (approved):
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/> 

> Reviewed 4 of 4 files at r1.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/52684#-:-MEYKDxD2d5B6_WZNvKc:bg7igkv)* status: :shipit: complete! 1 of 0 LGTMs obtained



<!-- Sent from Reviewable.io -->

otan committed:
geo: fix to_json representation of GeoJSON

Fixed the to_json() representation of geospatial objects to be the
same as the GeoJSON format. Added a test to ensure they
match with ST_AsGeoJSON.

Release note: None

otan commented:
bors r=sumeerbhola
